examples/pytorch/correct_and_smooth/model.py

```diff
@@ -0,0 +1,254 @@
+        nn.init.zeros_(self.bias)
+
+    def forward(self, x):
+        return x + self.bias
+
+
+class GAT(nn.Module):
+    def __init__(self,
+                 in_feats,
+                 n_classes,
+                 n_hidden,
+                 n_layers,
+                 n_heads,
+                 activation,
+                 dropout=0.0,
+                 attn_drop=0.0):
+        super().__init__()
+        self.in_feats = in_feats
+        self.n_classes = n_classes
+        self.n_hidden = n_hidden
+        self.n_layers = n_layers
+        self.num_heads = n_heads
+
+        self.convs = nn.ModuleList()
+        self.linear = nn.ModuleList()
+        self.bns = nn.ModuleList()
+
+        for i in range(n_layers):
+            in_hidden = n_heads * n_hidden if i > 0 else in_feats
+            out_hidden = n_hidden if i < n_layers - 1 else n_classes
+            out_channels = n_heads
+
+            self.convs.append(GATConv(in_hidden,
+                                      out_hidden,
+                                      num_heads=n_heads,
+                                      attn_drop=attn_drop))
+            self.linear.append(nn.Linear(in_hidden, out_channels * out_hidden, bias=False))
+            if i < n_layers - 1:
+                self.bns.append(nn.BatchNorm1d(out_channels * out_hidden))
+
+        self.bias_last = Bias(n_classes)
+
+        self.dropout0 = nn.Dropout(min(0.1, dropout))
+        self.dropout = nn.Dropout(dropout)
+        self.activation = activation
+
+    def forward(self, graph, feat):
+        h = feat
+        h = self.dropout0(h)
+
+        for i in range(self.n_layers):
+            conv = self.convs[i](graph, h)
+            linear = self.linear[i](h).view(conv.shape)
+
+            h = conv + linear
+
+            if i < self.n_layers - 1:
+                h = h.flatten(1)
+                h = self.bns[i](h)
+                h = self.activation(h)
+                h = self.dropout(h)
+
+        h = h.mean(1)
+        h = self.bias_last(h)
+
+        return F.log_softmax(h, dim=-1)
+
+class LabelPropagation(nn.Module):
+    r"""
+
+    Description
+    -----------
+    Introduced in `Learning from Labeled and Unlabeled Data with Label Propagation <https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.14.3864&rep=rep1&type=pdf>`_
+
+    .. math::
+        \mathbf{Y}^{\prime} = \alpha \cdot \mathbf{D}^{-1/2} \mathbf{A}
+        \mathbf{D}^{-1/2} \mathbf{Y} + (1 - \alpha) \mathbf{Y},
+
+    where unlabeled data is inferred by labeled data via propagation.
+
+    Parameters
+    ----------
+        num_layers: int
+            The number of propagations.
+        alpha: float
+            The :math:`\alpha` coefficient.
+    """
+    def __init__(self, num_layers, alpha):
+        super(LabelPropagation, self).__init__()
+
+        self.num_layers = num_layers
+        self.alpha = alpha
+    
+    @torch.no_grad()
+    def forward(self, g, labels, mask=None, post_step=lambda y: y.clamp_(0., 1.)):
```

Comment:
A1 and A2 can be different. For example, for MLP, A1 is DAD, A2 is DA: https://github.com/CUAI/CorrectAndSmooth/blob/b910314a59270984f5e249462ee3faa815fc9a0c/run_experiments.py#L140